File: UVa/Contents/c087/c087.c
#include<stdio.h>
#include<math.h>

int gcd(int, int);
int cal(int, int*);

int main(void)
{
	int n, temp;
	int set[51];
	
	while(1)
	{
		scanf("%d", &n);
		
		if(!n)
			break;
		
		for(int i = 0; i < n; i++)
			scanf("%d", &set[i]);
		
		temp = cal(n, set);
		if(!temp)
			printf("No estimate for this data set.\n");
		else
			printf("%lf\n", sqrt(6.0 * n * (n - 1) / 2 / temp));
	}
	
	return 0;
}

int gcd(int a, int b)
{
	if(!(a % b))
		return b;
	else
		return gcd(b, a % b);
}

int cal(int n, int arg[])
{
	int count = 0;
	
	for(int i = 0; i < n; i++)
	{
		for(int j = 0; j < n; j++)
		{
			if(i == j)
				continue;
			else if(gcd(arg[i], arg[j]) == 1)
				count++;
		}
	}
	
	return count / 2;
}

File: UVa/Contents/c087/c087.py
import sys

def gcd(a, b):
    if b == 0:
        return a
    else:
        return gcd(b, a % b)

for n in sys.stdin:
    n = int(n)

    if n == 0:
        break
    
    temp = []
    for i in range(n):
        temp.append(int(input()))
    
    count = 0
    for i in range(n):
        for j in range(n):
            if i == j:
                continue
            elif gcd(temp[i], temp[j]) == 1:
                count += 1

    count /= 2
    if count == 0:
        print("No estimate for this data set.")
    else:
        print("%.6f" % (6 * n * (n - 1) / 2 / count) ** 0.5)
